Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/UpdateDescription.swift

```diff
@@ -44,7 +55,34 @@ public final class UpdateDescription: Codable {
         }
 
         return updateDocument
-    }()
+    }
+
+    /**
+     Unilaterally merge an update description into this update description.
+     - parameter otherDescription: the update description to merge into this
+     - returns: this merged update description
+     */
+    @discardableResult
+    public func merge(with otherDescription: UpdateDescription) -> UpdateDescription {
+        self.updatedFields = self.updatedFields.filter {
+            !otherDescription.removedFields.contains($0.key)
+        }
+        self._removedFields = NSOrderedSet(array: self.removedFields.filter {
+            return !otherDescription.updatedFields.keys.contains($0)
+        })
+
+        otherDescription.updatedFields.forEach {
+            self.updatedFields[$0.key] = $0.value
+        }
+
+        guard let mutableCopy = self._removedFields.mutableCopy() as? NSMutableOrderedSet else {
```

Comment:
[nit] can you add a comment explaining what's happening with this section of remaining code?